lib/node_modules/@stdlib/lapack/base/dlange/lib/base.js

```diff
@@ -0,0 +1,204 @@
+*
+* @private
+* @param {string} norm - specifies the value to be returned
+* @param {NonNegativeInteger} M - number of rows of `A`
+* @param {NonNegativeInteger} N - number of columns of `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {PositiveInteger} offsetA - starting index for `A`
+* @param {Float64Array} work - workspace array
+* @param {integer} strideW - stride length for `work`
+* @param {PositiveInteger} offsetW - starting index for `work`
+* @returns {number} matrix norm
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var work = new Float64Array( 2 );
+*
+* var out = dlange( 'M', 2, 2, A, 2, 1, 0, work, 1, 0 );
+* // returns 4.0
+*/
+function dlange( norm, M, N, A, strideA1, strideA2, offsetA, work, strideW, offsetW ) {
+	var order;
+	var value;
+	var sum;
+	var tmp;
+	var out;
+	var sa0;
+	var sa1;
+	var sa;
+	var i;
+	var j;
+
+	norm = lowercase( norm );
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		// For row-major matrices, the last dimension has the fastest changing index...
+		order = 'row-major';
+		sa0 = strideA2; // stride for innermost loop
+		sa1 = strideA1; // stride for outermost loop
+	} else { // order == 'column-major'
+		// For column-major matrices, the first dimension has the fastest changing index...
+		order = 'column-major';
+		sa0 = strideA1;
+		sa1 = strideA2;
+	}
+
+	if ( min( M, N ) === 0.0 ) {
+		value = 0.0;
+	} else if ( norm === 'm' ) {
+		// Find max( abs( A[ i, j ] ) )
+		value = 0.0;
+		if ( order === 'column-major' ) {
+			for ( j = 0; j < N; j++ ) {
+				sa = offsetA + ( j * sa0 );
+				for ( i = 0; i < M; i++ ) {
+					tmp = abs( A[ sa + ( i * sa1 ) ] );
+					if ( value <= tmp || isnan( tmp ) ) {
+						value = tmp;
+					}
+				}
+			}
+			return value;
+		}
+		// order === 'row-major'
+		for ( j = 0; j < M; j++ ) {
+			sa = offsetA + ( j * sa1 );
+			for ( i = 0; i < N; i++ ) {
+				tmp = abs( A[ sa + ( i * sa0 ) ] );
+				if ( value <= tmp || isnan( tmp ) ) {
+					value = tmp;
+				}
+			}
+		}
+		return value;
+	}
+	if ( norm === 'o' || norm === '1' ) {
+		// Find norm1( A )
+		value = 0.0;
+		if ( order === 'column-major' ) {
+			for ( j = 0; j < N; j++ ) {
+				sum = 0.0;
+				sa = offsetA + ( j * sa1 );
+				for ( i = 0; i < M; i++ ) {
+					sum += abs( A[ sa + ( i * sa0 ) ] );
+				}
+				if ( value < sum || isnan( sum ) ) {
+					value = sum;
+				}
+			}
+			return value;
+		}
+		// order === 'row-major'
+		// For computing one norm, first dimension has the fastest changing index...
+		sa0 = strideA1;
+		sa1 = strideA2;
+		for ( j = 0; j < N; j++ ) {
+			sum = 0.0;
+			sa = offsetA + ( j * sa1 );
+			for ( i = 0; i < M; i++ ) {
+				sum += abs( A[ sa + ( i * sa0 ) ] );
+			}
+			if ( value < sum || isnan( sum ) ) {
+				value = sum;
+			}
+		}
```

Comment:
And if the above comment is correct, this implementation is same as `column-major` and we can just remove this loop.